common/beerocks/bwl/include/bwl/key_value_parser.h

```diff
@@ -0,0 +1,66 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * Copyright (c) 2016-2019 Intel Corporation
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#ifndef _BWL_KEY_VALUE_PARSER_H_
+#define _BWL_KEY_VALUE_PARSER_H_
+
+#include <list>
+#include <string>
+#include <unordered_map>
+
+namespace bwl {
+
+/**
+ * @brief On KeyValueParser there are two types of object that can be parsed:
+ * parsed_line_t - Shall be used on key-value data with unique keys for each element, seperated by
+ * a space or a new line (Default seperator, can be modified on use). For example:
+ * "key1=value1 key2=value2 key3=value3 ..."
+ * --Or--
+ * "key1=value1
+ *  key2=value2
+ *  key3=value3
+ *  ..."
+ * 
+ * parsed_multiline_t - Shall be used in case there are 2 or more "parsed_line_t" which each "line"
+ * could have the same keys as the previous one. For example: 
+ * "key1=Line1value1 key2=Line1value2 key3=Line1value3 ...
+ *  key1=Line2value1 key2=Line2value2 key3=Line2value3 ...
+ *  key1=Line3value1 key2=Line3value2 key3=Line3value3 ...
+ *  ...""
+ */
+
+typedef std::unordered_map<std::string, std::string> parsed_line_t;
+typedef std::list<parsed_line_t> parsed_multiline_t;
```

Comment:
Can't you replace those two with a single `std::unordered_multimap`?

Reply:
No. Some of the hostapd dumps are repeating the same keys (ACS report for example).
So I have to create a list of `o with a single `std::unordered_multimap`.

Reply:
Isn't the idea of `std::unordered_multimap` is to be able to multiple elements for the same key?

Reply:
Yes, but all the values on a single list entry are related only to values on the entry and not another entry.
If it was multimap we could not know who belongs to who.
For example:
`
ch=36 bw=80
ch=38 bw=20
`
`bw=80` belongs only to channel 36. If it was multimap you couldn't know that.

Reply:
I agree with Moran on this one.